addons/addon-base-raas/packages/base-raas-rest-api/lib/controllers/migration-controller.js

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+async function configure(context) {
+  const router = context.router();
+  const wrap = context.wrap;
+
+  // ===============================================================
+  //  PUT /mystudies (mounted to /api/migrate)
+  // ===============================================================
+  router.put(
+    '/mystudies',
+    wrap(async (req, res) => {
+      const migrationService = await context.service('migrationService');
+      const requestContext = res.locals.requestContext;
+      const migrationMappings = req.body;
+      const result = await migrationService.migrateMyStudiesPermissions(requestContext, migrationMappings);
+      res.status(200).json(result);
+    }),
+  );
+
+  // ===============================================================
+  //  GET /mystudies (mounted to /api/migrate)
+  // ===============================================================
+  router.get(
+    '/mystudies',
+    wrap(async (req, res) => {
+      const migrationService = await context.service('migrationService');
+      const requestContext = res.locals.requestContext;
+      const result = await migrationService.listMyStudies(requestContext);
```

Comment:
Does the response make it evident which studies require migration? These values in the response body will help a lot for admins to figure out what to migrate:

```
studyId: <sampleStudyId>,
uid: <sampleUidValue>,
authProvider: <respectiveAuthProviderId / 'internal'> 

```